inngest-spring-boot-demo/build.gradle.kts

```diff
@@ -39,6 +45,9 @@ dependencyManagement {
 tasks.withType<Test> {
     useJUnitPlatform()
     systemProperty("junit.jupiter.execution.parallel.enabled", true)
+    systemProperty("test-group", "unit-test")
+
+    jvmArgs = listOf("-Dnet.bytebuddy.experimental=true")
```

Comment:
what does this do?

Reply:
my bad i was going to mention it and forgot, `bytebuddy` is a dependency for dynamically generating/altering classes at runtime. It's used by the env variable mocking library `system-stubs-jupiter`.

Without enabling experimental support it throws this error:
```
Caused by:
            java.lang.IllegalArgumentException: Java 21 (65) is not supported by the current version of Byte Buddy which officially supports Java 20 (64) - update Byte Buddy or set net.bytebuddy.experimental as a VM property
                at net.bytebuddy.utility.OpenedClassReader.of(OpenedClassReader.java:96)
                at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:4011)
                at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2224)
                at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$UsingTypeWriter.make(DynamicType.java:4050)
                at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3734)
                at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$Delegator.make(DynamicType.java:3986)
                at uk.org.webcompere.systemstubs.environment.EnvironmentVariableMocker.<clinit>(EnvironmentVariableMocker.java:42)
```
https://github.com/raphw/byte-buddy/issues/1396

It's fixed in later versions which means that it will also run if I force use a newer version with `implementation("net.bytebuddy:byte-buddy:1.15.1")`, but i'm not sure if that breaks the original dependency that pulls the `1.13` version. Do you know of a tool that draws a graph of dependencies? Although i believe it's tricky since dependency version will also depend on the currently used JVM version.

Reply:
Okay so the dependency graph (had to manually check lol) is:
```
spring-boot-starter-test:2.7.28
└── mockito-core:4.5.1
    └── byte-buddy:1.12.9
```
I think `spring-boot-starter-test` version is stuck to v2 because of the JVM 8 requirement.

Reply:
```java
Execution failed for task ':inngest-spring-boot-demo:compileTestJava'.
> Could not resolve all files for configuration ':inngest-spring-boot-demo:testCompileClasspath'.
   > Could not resolve org.springframework.boot:spring-boot-starter-test:3.3.0.
     Required by:
         project :inngest-spring-boot-demo
      > Dependency resolution is looking for a library compatible with JVM runtime version 8, but 'org.springframework.boot:spring-boot-starter-test:3.3.0' is only compatible with JVM runtime version 17 or newer.
```

Reply:
ok cool thanks for the background. Maybe add a comment here that this is required by `system-stubs-jupiter`?